src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleTypeSymbol.vb

```diff
@@ -37,203 +33,152 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
-            New WellKnownMember() {WellKnownMember.System_ValueTuple_T4__Item1, WellKnownMember.System_ValueTuple_T4__Item2, WellKnownMember.System_ValueTuple_T4__Item3, WellKnownMember.System_ValueTuple_T4__Item4},
-            New WellKnownMember() {WellKnownMember.System_ValueTuple_T5__Item1, WellKnownMember.System_ValueTuple_T5__Item2, WellKnownMember.System_ValueTuple_T5__Item3, WellKnownMember.System_ValueTuple_T5__Item4, WellKnownMember.System_ValueTuple_T5__Item5},
-            New WellKnownMember() {WellKnownMember.System_ValueTuple_T6__Item1, WellKnownMember.System_ValueTuple_T6__Item2, WellKnownMember.System_ValueTuple_T6__Item3, WellKnownMember.System_ValueTuple_T6__Item4, WellKnownMember.System_ValueTuple_T6__Item5, WellKnownMember.System_ValueTuple_T6__Item6},
-            New WellKnownMember() {WellKnownMember.System_ValueTuple_T7__Item1, WellKnownMember.System_ValueTuple_T7__Item2, WellKnownMember.System_ValueTuple_T7__Item3, WellKnownMember.System_ValueTuple_T7__Item4, WellKnownMember.System_ValueTuple_T7__Item5, WellKnownMember.System_ValueTuple_T7__Item6, WellKnownMember.System_ValueTuple_T7__Item7},
-            New WellKnownMember() {WellKnownMember.System_ValueTuple_TRest__Item1, WellKnownMember.System_ValueTuple_TRest__Item2, WellKnownMember.System_ValueTuple_TRest__Item3, WellKnownMember.System_ValueTuple_TRest__Item4, WellKnownMember.System_ValueTuple_TRest__Item5, WellKnownMember.System_ValueTuple_TRest__Item6, WellKnownMember.System_ValueTuple_TRest__Item7, WellKnownMember.System_ValueTuple_TRest__Rest}}
-
-        Public Overrides ReadOnly Property IsTupleType As Boolean
-            Get
-                Return True
-            End Get
-        End Property
+        Friend Const RestPosition = 8
+        Friend Const RestIndex = RestPosition - 1
+
+        Friend Const TupleTypeName = "ValueTuple"
+        Friend Const RestFieldName = "Rest"
+
+        Private Shared ReadOnly tupleTypes As WellKnownType() =
+            {
+                WellKnownType.System_ValueTuple_T1, WellKnownType.System_ValueTuple_T2, WellKnownType.System_ValueTuple_T3, WellKnownType.System_ValueTuple_T4,
+                WellKnownType.System_ValueTuple_T5, WellKnownType.System_ValueTuple_T6, WellKnownType.System_ValueTuple_T7, WellKnownType.System_ValueTuple_TRest
+            }
+
+        Private Shared ReadOnly tupleCtors As WellKnownMember() =
+            {
+                WellKnownMember.System_ValueTuple_T1__ctor, WellKnownMember.System_ValueTuple_T2__ctor, WellKnownMember.System_ValueTuple_T3__ctor,
+                WellKnownMember.System_ValueTuple_T4__ctor, WellKnownMember.System_ValueTuple_T5__ctor, WellKnownMember.System_ValueTuple_T6__ctor,
+                WellKnownMember.System_ValueTuple_T7__ctor, WellKnownMember.System_ValueTuple_TRest__ctor
+            }
+
+        Private Shared ReadOnly tupleMembers As WellKnownMember()() =
+            {
+                ({WellKnownMember.System_ValueTuple_T1__Item1}),
```

Comment:
And the current does? 😕 It's like C# as put through a code convertor, hence following c# initialization syntax. 
It is written in VB.net would not exploit it's features and capacities.  Eg Collection initializer syntax
The alignment could be better true, but it is way better to look at and read than the existing. Admit it.